src/file.cc

```diff
@@ -63,24 +63,33 @@ std::pair<StringView, StringView> split_path(StringView path)
     return { {path.begin(), slash+1}, {slash+1, path.end()} };
 }
 
+struct MallocDeleter
+{
+    void operator()(char *ptr) const
+    {
+        free(ptr);
+    }
+};
+
 String real_path(StringView filename)
 {
     if (filename.empty())
         return {};
 
-    char buffer[PATH_MAX+1];
-
     StringView existing = filename;
     StringView non_existing{};
 
+    using unique_cstr = std::unique_ptr<char, MallocDeleter>;
```

Comment:
Kakoune moved away to a custom `UniquePtr` implementation for compilation speed reasons, it does not support custom deleters as we did not have actual uses, but in this specific case a `OnScopeEnd free{[&] { free(res); };` would probably do the trick.